[tactics] Refine test for unresolved evars: not reachable from initial evars (#7825)
coq/coq: doc/changelog/02-specification-language/07825-rechable-from-evars.rst

> - In `refine`, new existential variables unified with existing ones are no

Reviewer: ```suggestion
- **Changed:**
  In :tacn:`refine`, new existential variables unified with existing ones are no
```

>   longer considered as fresh. The behavior of `simple refine` no longer depend on

Reviewer: ```suggestion
  longer considered as fresh. The behavior of :tacn:`simple refine` no longer depends on
```

>   the orientation of evar-evar unification problems, new existential variables

Reviewer 2: ```suggestion
  the orientation of evar-evar unification problems, and new existential variables
```

Also, is "orientation" the right word here?  I wasn't sure what was meant, but that may just be my lack of knowledge.

Reviewer 3: I think "orientation" is correct; to me it means "left-to-right" vs "right-to-left", i.e., the difference between `unify x y` and `unify y x` when you need to bias one side over the other (e.g., if you're unifying two evars and need to decide which one gets filled with the other)

Reviewer 2: "left-to-right" and "right-to-left" suggest "direction" as a useful wording.  Your choice.
